burotel/indico_burotel/cli.py

```diff
@@ -0,0 +1,151 @@
+               format(field, old, new))
+    print
+
+
+@cli.command()
+@click.argument('csv_file', type=click.File('rb'))
+@click.option('--dry-run', is_flag=True, help="Don't actually change the database, just report on the changes")
+def update(csv_file, dry_run):
+    """Update the Burotels from a CSV file."""
+    num_changes = 0
+    r = csv.reader(csv_file)
+
+    for room_id, division, building, floor, number, verbose_name, owner_email, egroup, action in r:
+        owner = get_owner_from_email(owner_email)
+        data = {
+            'id': int(room_id.decode('utf-8-sig')) if room_id else None,
+            'division': division,
+            'building': building,
+            'floor': floor,
+            'number': number,
+            'verbose_name': verbose_name,
+            'owner': owner,
+            'acl_entries': {owner, GroupProxy(egroup, 'cern-ldap')},
+            'action': action or 'UPDATE'
+        }
+        if not data['id'] and action != 'ADD':
+            print cformat("%{yellow}! Only ADD lines can have an empty Desk ID. Ignoring line.")
+            continue
+        elif data['action'] == 'UPDATE':
+            room = Room.get(room_id)
+            if not room:
+                print cformat("%{yellow}! Desk with ID {} not found.").format(room_id)
+                continue
+            changes = check_changed_fields(room, data)
+            if changes:
+                num_changes += 1
+                _print_changes(room, changes)
+                if not dry_run:
+                    change_room(room, changes)
+        elif data['action'] == 'ADD':
+            existing_room = Room.query.filter(Room.building == building,
+                                              Room.floor == floor,
+                                              Room.number == number,
+                                              Room.verbose_name == verbose_name).first()
+            if existing_room:
+                # a room with the exact same designation already exists
+                print (cformat("%{yellow}!%{reset} A desk with the name %{cyan}{}/{}-{} {}%{reset} already exists")
+                       .format(building, floor, number, verbose_name))
+                continue
+            print cformat("%{green!}+%{reset} New desk %{cyan}{}/{}-{} {}").format(
+                building, floor, number, verbose_name)
+            if not dry_run:
+                room = Room(building=building, floor=floor, number=number, division=division,
+                            verbose_name=verbose_name, owner=owner, location=get_location(building),
+                            protection_mode=ProtectionMode.protected)
+                db.session.add(room)
+                if egroup:
+                    room.set_attribute_value('manager-group', egroup)
+        elif data['action'] == 'REMOVE':
+            room = Room.get(room_id)
+            print cformat("%{red}-%{reset} {r.building}/{r.floor}-{r.number} {r.verbose_name}").format(r=room)
```

Comment:
`r.name` instead of building the name manually?

Reply:
Indeed.